server/minestom/src/main/java/net/labymod/serverapi/server/minestom/LabyModProtocolService.java

```diff
@@ -0,0 +1,143 @@
+package net.labymod.serverapi.server.minestom;
+
+import net.labymod.serverapi.api.Protocol;
+import net.labymod.serverapi.api.logger.NoOpProtocolPlatformLogger;
+import net.labymod.serverapi.api.logger.ProtocolPlatformLogger;
+import net.labymod.serverapi.api.packet.Packet;
+import net.labymod.serverapi.api.payload.PayloadChannelIdentifier;
+import net.labymod.serverapi.api.payload.io.PayloadReader;
+import net.labymod.serverapi.api.payload.io.PayloadWriter;
+import net.labymod.serverapi.core.AbstractLabyModProtocolService;
+import net.labymod.serverapi.core.packet.serverbound.login.VersionLoginPacket;
+import net.labymod.serverapi.server.common.AbstractServerLabyModProtocolService;
+import net.labymod.serverapi.server.common.JavaProtocolLogger;
+import net.labymod.serverapi.server.common.model.player.AbstractServerLabyModPlayer;
+import net.labymod.serverapi.server.minestom.event.LabyModInstalledAddonsUpdateEvent;
+import net.labymod.serverapi.server.minestom.event.LabyModPacketReceivedEvent;
+import net.labymod.serverapi.server.minestom.event.LabyModPacketSentEvent;
+import net.labymod.serverapi.server.minestom.handler.DefaultVersionLoginPacketHandler;
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.entity.Player;
+import net.minestom.server.event.player.PlayerDisconnectEvent;
+import net.minestom.server.event.player.PlayerPluginMessageEvent;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+import java.util.UUID;
+
+public class LabyModProtocolService extends AbstractServerLabyModProtocolService<LabyModPlayer> {
+
+    private static final LabyModProtocolService INSTANCE = new LabyModProtocolService();
+    private final ProtocolPlatformLogger logger = NoOpProtocolPlatformLogger.get();
+    private boolean initialized = false;
+
+    private LabyModProtocolService() {
+    }
+
+    public static LabyModProtocolService get() {
+        return INSTANCE;
+    }
+
+    public void init() {
```

Comment:
please make this private and create a static method called `initialize` that calls this to keep the naming across server implementations consistent.

Alternatively, this could be private and called in the constructor, as no plugin instance is needed

Reply:
Fixed in latest